nornir/plugins/inventory/netbox.py

```diff
@@ -143,10 +147,66 @@ def __init__(
             "NB_TOKEN", "0123456789abcdef0123456789abcdef01234567"
         )
 
+        # Load in user private key from environment var or file load
+        nb_private_key = nb_private_key or os.environ.get("NB_PRIVATE_KEY", None)
+
+        if os.environ.get('NB_PRIVATE_KEY') is not None:
+            nb_private_key = os.environ.get('NB_PRIVATE_KEY')
+        elif os.environ.get('NB_PRIVATE_KEY') is None and nb_private_key:
+            with open(nb_private_key, 'r') as f:
+                nb_private_key = f.read()
+
```

Comment:
This seems to be kind of redundant with line 151.
nb_private_key is None by default and should be set to nb_private_key if that argument is provided, else set to ENV variable if that is set.

This if/elif can just be combined into a single if statement read from it:
```
if nb_private_key:
    read_from_file()
```

You could also consider making nb_private_key a ```Union[str, Path]``` so that you can test if it is a string or a Path object, if a Path object is passed & is a file read it, if it is a string assign it to nb_private_key.

Reply:
old code, forgot to remove it. Also updated the conditional to be as you said.

Reply:
I think the "old" line 152 could remain, since it clearly sets the nb_private_key var to the argument pass to the function or the env variable. Depending on the point above (can it be a path a file, or a string containing the key) you can then build logic to read the file or leave as is (in case of just the key being passed).